misc/python/materialize/postgres_consistency/ignore_filter/pg_inconsistency_ignore_filter.py

```diff
@@ -0,0 +1,203 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+from materialize.output_consistency.enum.enum_constant import EnumConstant
+from materialize.output_consistency.execution.evaluation_strategy import (
+    EvaluationStrategyKey,
+)
+from materialize.output_consistency.expression.expression import Expression
+from materialize.output_consistency.expression.expression_characteristics import (
+    ExpressionCharacteristics,
+)
+from materialize.output_consistency.expression.expression_with_args import (
+    ExpressionWithArgs,
+)
+from materialize.output_consistency.ignore_filter.inconsistency_ignore_filter import (
+    IgnoreVerdict,
+    InconsistencyIgnoreFilter,
+    NoIgnore,
+    PostExecutionInconsistencyIgnoreFilterBase,
+    PreExecutionInconsistencyIgnoreFilterBase,
+    YesIgnore,
+)
+from materialize.output_consistency.input_data.return_specs.date_time_return_spec import (
+    DateTimeReturnTypeSpec,
+)
+from materialize.output_consistency.input_data.types.date_time_types_provider import (
+    TIME_TYPE_IDENTIFIER,
+)
+from materialize.output_consistency.operation.operation import DbFunction, DbOperation
+from materialize.output_consistency.query.query_result import QueryFailure
+from materialize.output_consistency.query.query_template import QueryTemplate
+from materialize.output_consistency.validation.validation_message import ValidationError
+
+
+class PgInconsistencyIgnoreFilter(InconsistencyIgnoreFilter):
+    """Allows specifying and excluding expressions with known output inconsistencies"""
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.pre_execution_filter = PgPreExecutionInconsistencyIgnoreFilter()
+        self.post_execution_filter = PgPostExecutionInconsistencyIgnoreFilter()
+
+
+class PgPreExecutionInconsistencyIgnoreFilter(
+    PreExecutionInconsistencyIgnoreFilterBase
+):
+    def _matches_problematic_function_invocation(
+        self,
+        db_function: DbFunction,
+        expression: ExpressionWithArgs,
+        all_involved_characteristics: set[ExpressionCharacteristics],
+    ) -> IgnoreVerdict:
+        inconsistent_value_ordering_functions = ["array_agg", "string_agg"]
```

Comment:
JFYI `ORDER BY` is supported as an argument to those functions.

Reply:
✅ Tightened the ignore pattern.